File: CSharp/HtmlWalker/Tag.Base/ContentTextTag.cs
using HtmlWalker.Platform;

namespace HtmlWalker.Tag.Base
{
    /// <summary>
    ///   This class is used to support implementation tag adapters for HTML tags that only have <see cref="ITag.TextValue"/> content
    /// </summary>
    /// <author>Robert R Flores</author>
    public abstract class ContentTextTag : BaseTag
    {
        /// <summary>
        ///   Instantiates a new content text tag
        /// </summary>
        /// <param name="factory">used to produce objects as per the specific platform</param>
        /// <param name="tag">object housed by this tag adapter</param>
        public ContentTextTag(WalkerPlatform.WalkerFactory factory, object tag) : base(factory, tag) { }

        /// <summary>
        ///   Implement <see cref="ITag.ContentType"/>
        /// </summary>
        /// <returns><see cref="ContentType.TEXT"/></returns>
        public override ContentType ContentType => ContentType.TEXT;

        /// <summary>
        ///   Implement <see cref="ITag.TextValue"/>
        /// </summary>
        public override string TextValue { get => Api.GetText(Tag); set => Api.SetText(Tag, value); }

        /// <summary>
        ///   Impleents <see cref="ITag.OwnedTags"/>
        /// </summary>
        public override TagList OwnedTags => null;

        /// <summary>
        ///   Impleents <see cref="ITag.OwnedText"/>
        /// </summary>
        public override string OwnedText => TextValue;
    }
}

File: Java/htmlwalker/src/main/java/htmlwalker/tag/base/ContentTextTag.java
/*
 * 
 */
package htmlwalker.tag.base;

import htmlwalker.platform.*;

/**
 * This class is used to support implementation tag adapters for HTML tags that only have text content.
 * 
 * @author Robert Flores
 */
public abstract class ContentTextTag extends BaseTag
{
	
	/**
	 * Instantiates a new content text tag.
	 *
	 * @param factory - used to produce objects as per the specific platform
	 * @param tag - object housed by this tag adapter
	 */
	protected ContentTextTag(WalkerFactory factory, Object tag) { super(factory, tag); }
	
	/**
	 * @see HtmlWalker.ITag#contentType()
	 */
	@Override
	public ContentType contentType() { return ContentType.TEXT; }
	
	/**
	 * @see HtmlWalker.ITag#getTextValue()
	 */
	@Override
	public String getTextValue() { return this.api.getText(tag()); }
	
	/**
	 * @see HtmlWalker.ITag#setTextValue(java.lang.String)
	 */
	@Override
	public void setTextValue(String text) { this.api.setText(tag(), text); }
	
	/**
	 * @see HtmlWalker.ITag#ownedTags()
	 */
	@Override
	public TagList ownedTags() { return null; }
	
	@Override
	public String ownedText() { return getTextValue(); }
}
